src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -2204,6 +2443,381 @@ def construct_command(self, task_exec, task_args, task_numcores,
         return poe_command, None
 
 
+# ==============================================================================
+#
+# The Launch Method Implementation for Running YARN applications
+#
+class LaunchMethodYARN(LaunchMethod):
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg, logger):
+
+        LaunchMethod.__init__(self, cfg, logger)
+
+
+    # --------------------------------------------------------------------------
+    #
+    @classmethod
+    def lrms_config_hook(cls, name, cfg, lrms, logger):
+        """
+        FIXME: this config hook will inspect the LRMS nodelist and, if needed,
+               will start the YRN cluster on node[0].
+        """
+
+        logger.info('Hook called by YARN LRMS with the name %s'%lrms.name)
+
+        def config_core_site(node):
+
+            core_site_file = open(os.getcwd()+'/hadoop/etc/hadoop/core-site.xml','r')
+            lines = core_site_file.readlines()
+            core_site_file.close()
+
+            prop_str  = '<property>\n'
+            prop_str += '  <name>fs.default.name</name>\n'
+            prop_str += '    <value>hdfs://%s:54170</value>\n'%node
```

Comment:
The agent is not always  running on a compute note.  In fact, that is an exception at the moment...  In general it might be a head-, mom- or compute node.  Even on compute nodes it is not necessarily safe to assume that they are single user (even though that it is usually the case...).  Mom-nodes are also multi-user...